instrumentation/opentelemetry_oban/lib/opentelemetry_oban/plugin_handler.ex

```diff
@@ -63,4 +66,44 @@ defmodule OpentelemetryOban.PluginHandler do
 
     OpentelemetryTelemetry.end_telemetry_span(@tracer_id, metadata)
   end
+
+  defp end_span_plugin_attrs(%{plugin: Oban.Plugins.Cron} = metadata) do
+    %{jobs_count: length(metadata[:jobs])}
+  end
+
+  defp end_span_plugin_attrs(%{plugin: Oban.Plugins.Gossip} = metadata) do
+    Map.take(metadata, [:gossip_count])
+  end
+
+  defp end_span_plugin_attrs(%{plugin: Oban.Plugins.Lifeline} = metadata) do
+    Map.take(metadata, [:discarded_count, :rescued_count])
+  end
+
+  defp end_span_plugin_attrs(%{plugin: Oban.Plugins.Pruner} = metadata) do
+    Map.take(metadata, [:pruned_count])
+  end
+
+  defp end_span_plugin_attrs(%{plugin: Oban.Pro.Plugins.DynamicCron} = metadata) do
+    %{jobs_count: length(metadata[:jobs])}
+  end
+
+  defp end_span_plugin_attrs(%{plugin: Oban.Pro.Plugins.DynamicLifeline} = metadata) do
+    Map.take(metadata, [:discarded_count, :rescued_count])
+  end
+
+  defp end_span_plugin_attrs(%{plugin: Oban.Pro.Plugins.DynamicPrioritizer} = metadata) do
+    Map.take(metadata, [:reprioritized_count])
+  end
+
+  defp end_span_plugin_attrs(%{plugin: Oban.Pro.Plugins.DynamicPruner} = metadata) do
+    Map.take(metadata, [:pruned_count])
+  end
+
+  defp end_span_plugin_attrs(%{plugin: Oban.Pro.Plugins.DynamicScaler} = metadata) do
+    Map.take(metadata[:scaler], [:last_scaled_to, :last_scaled_at])
+  end
```

Comment:
Please verify if this plugin if you can, they were other more rich objects inside, but I am not sure how to go about it.

I havent test it myself, I never used this plugin before.